### PR TITLE
modelapi: provide more options to decide how model_filter works

### PR DESCRIPTION
**Pull request by villelaitila**
When executing model_filter for elem e with "Ignore" mode, it ignores elements that are external to e.

"Direct" mode changes this behavior: it picks also external elements associated with descendants of e. However, it is limited to direct assocations.

"DirectAndIndirect" mode selects all directly and indirectly associated elements and their descendants.

Moved tests to main directory from src.

### CODE REVIEW

File: tests/sgraph/modelapi_test.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    elem = subgraph1.findElementFromPath('/used-directly-from-nginx/src/used-directly-from-nginx.c/child/child-of-child/child-of-child-of-child')
+    assert None is not elem
+    assert 'testvalue1' == elem.attrs['test_attribute1']
+
+
+
+def test_filter_model_with_indirect():
+    model, model_api = get_model_and_model_api('modelfile.xml')
+    # Check that model root node has two children nginx and foo
```
Comment (skis84) [inline tests/sgraph/modelapi_test.py:42]: This comment does not seem relevant in this test case, there is no checking of both nginx and foo

File: tests/sgraph/modelapi_test.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    elem1 = model.createOrGetElementFromPath('/nginx')
+    subgraph1 = model_api.filter_model(elem1, model,
+                                       filter_outgoing=FilterAssocations.DirectAndIndirect,
+                                       filter_incoming=FilterAssocations.DirectAndIndirect)
+    # Subgraph should now have nginx but not foo
```
Comment (skis84) [inline tests/sgraph/modelapi_test.py:47]: Comment seems irrelevant here too

File: tests/sgraph/modelapi_test.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    assert None is not elem
+    assert 'testvalue2' == elem.attrs['test_attribute2']
+
+def test_filter_model_with_ignore_param():
+    model, model_api = get_model_and_model_api('modelfile.xml')
+    # Check that model root node has two children nginx and foo
+    elem1 = model.createOrGetElementFromPath('/nginx')
+    subgraph1 = model_api.filter_model(elem1, model, filter_outgoing=FilterAssocations.Ignore,
+                                       filter_incoming=FilterAssocations.Ignore)
+    # Subgraph should now have nginx but not foo
```
Comment (skis84) [inline tests/sgraph/modelapi_test.py:65]: Irrelevant comments?

File: tests/sgraph/modelapi_test.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    model, model_api = get_model_and_model_api('modelfile.xml')
+    nginxc_element = model.createOrGetElementFromPath(f'{MODEL_PATH}nginx.c')
+    subgraph = model_api.filter_model(nginxc_element, model)
+    nginxh_element = subgraph.createOrGetElementFromPath(f'{MODEL_PATH}nginx.h')
+    assert 1 == len(nginxh_element.children)
+    assert 'testname' == nginxh_element.children[0].name
+    assert 'testtype' == nginxh_element.children[0].attrs['type']
+
+
+def test_filter_model_keeps_children_of_outgoing_dependency_2():
```
Comment (skis84) [inline tests/sgraph/modelapi_test.py:79]: Use a descriptive test case name instead of _2 in the end

**Review by skis84 — CHANGES_REQUESTED** [review]
Looks nice, some comments about the tests

File: tests/sgraph/modelapi_test.py (hunk 1 above)
Comment (villelaitila) [reply tests/sgraph/modelapi_test.py:42]: good point

File: tests/sgraph/modelapi_test.py (hunk 3 above)
Comment (villelaitila) [reply tests/sgraph/modelapi_test.py]: now cleaned

**Review by skis84 — APPROVED** [review]
